File: src/features/typing/data/java/subarray_sum_equals_k.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        count.put(0, 1);
        int presum = 0;
        int total = 0;
        for (int num : nums) {
            presum += num;
            total += count.getOrDefault(presum - k, 0);
            count.put(presum, count.getOrDefault(presum, 0) + 1);
        }
        return total;
    }
}

File: src/features/typing/data/dart/subarray_sum_equals_k.dart
class Solution {
  int subarraySum(List<int> nums, int k) {
    final count = <int, int>{};
    count[0] = 1;
    int presum = 0;
    int total = 0;
    for (final num in nums) {
      presum += num;
      total += count[presum - k] ?? 0;
      count[presum] = (count[presum] ?? 0) + 1;
    }
    return total;
  }
}
